pydiction.py

```diff
@@ -177,7 +177,10 @@ def get_yesno(msg="[Y/n]?"):
 
     """
     while True:
-        answer = raw_input(msg)
+        if sys.version_info[0] == 3:
```

Comment:
I find it better if the results of the version comparison were cached somewhere, e.g.

```python
text_getter = input if sys.version_info[0] == 3 else raw_input
while True:
    answer = text_getter(msg)
    # do stuff
```

This may not be a hot spot in the code but it serves to make it slightly faster and also cleaner, IMHO. Another approach is to "inject" the correct ``(raw_)input`` function at "compile-time" into the ``get_yesno`` function using keyword arguments, e.g.

```python
def get_yesno(msg="[Y/n]?", text_getter=(...)):
    pass
```

The second approach has the advantage that the expressions are only evaluated once as opposed to per-loop-iteration in your original suggestion or per-function-call in the first suggestion.